Handles array with join-type, schemas with only join-types and schemas with only pattern properties

Totoketchup opened the pull request:
I'm currently building a schema validator and wanted to use jsonschema2md in order to generated its documentation.

1 - Some of my schemas are just defined by join type properties as follow:

```json

{
    "$schema": "http://json-schema.org/draft-07/schema#",
    "$id": "...",
    "title": "...",
    "type": "object",
    "meta:status": "V2.0.0",
    "oneOf": [
        {
            "$ref": "continuous.json"
        },
        {
            "$ref": "equal.json"
        },
        {
            "$ref": "in_periodic.json"
        },
        {
            "$ref": "multi_enum.json"
        }
    ]
}

```

Commit 8f91237 solves this issue.

2 - Some of my schemas are as well only defined by `patternProperties`:

```json
{
    "$schema": "http://json-schema.org/draft-07/schema#",
    "$id": "...",
    "title": "....",
    "description": "...",
    "meta:status": "V2.0.0",
    "type": "object",
    "patternProperties": {
        "^[A-Za-z_][A-Za-z0-9_]*$": {
            "required": [
                "type"
            ],
            "type": "object",
            "oneOf": [
                {
                    "$ref": "ref1.json"
                },
                {
                    "$ref": "ref2.json"
                }
            ]
        }
    },
    "additionalProperties": false
}
```

Commit 52da75d solves this issue.

3 - Some of schemas contains array with `join-type` items:

```json

"children": {
            "type": "array",
            "items": {
                "oneOf": [
                    {
                        "$ref": "leaf.json"
                    },
                    {
                        "$ref": "node.json"
                    }
                ]
            },
            "minItems": 1
        }

```

Commit 930332f solves this issue

trieloff commented:
@Totoketchup thank you for your contribution. In order to merge this, I'll need three things:

1. the build/test failures should be fixed
2. you should provide test cases for your examples
3. and we'd like you to sign the Adobe CLA: http://opensource.adobe.com/cla.html

thank you!

Totoketchup commented:
I'm on it, thank you 👍

Totoketchup commented:
Hi @trieloff !
I added the tests on every commit.
On the first commit I fixed one bug you had on the tests where examples could be printed in a random order. So I ordered them according to their names to have a non random order due to the asynchronous promises.

Moreover, I noticed that most of the json.schema are defined with an 'allOf' property so I had to adjust the tests accordingly. But I don't really like how these latter are rendered because there is no $linkVal or $linkPath associated. What do you think would be a good strategy for those ?
